includes/polling/ucd-mib.inc.php

```diff
@@ -134,16 +134,11 @@
 
 $snmpdata = snmp_get_multi($device, ['memTotalSwap.0', 'memAvailSwap.0', 'memTotalReal.0', 'memAvailReal.0', 'memSysAvail.0', 'memTotalFree.0', 'memShared.0', 'memBuffer.0', 'memCached.0'], '-OQUs', 'UCD-SNMP-MIB');
 
-if (isset($snmpdata[0]['memSysAvail'])) {
+if (is_array($snmpdata[0])) {
     [$memTotalSwap, $memAvailSwap, $memTotalReal, $memAvailReal, $memSysAvail, $memTotalFree, $memShared, $memBuffer, $memCached] = $snmpdata[0];
```

Comment:
I suggest just removing the temporary variables and accessing the array by index instead.

Reply:
sounds good to me.

one follow up question, in the case there is the additional memSysAvail field, how do we make the change to use it in the graph. I've take a look includes/html/graphs/device/ucd_memory.inc.php and it doesn't seem to have condition check, will it cause error adding a non-exist field to rrd_options?

Reply:
You can add it to the end of the rrd defintion and I think it will still update the over values fine.  Then only newly created mibs will get the value.